### DISCUSSION TITLE
Tabs in ListResource

### DISCUSSION THREAD

**Discussion by antoinemy** [filamentphp/filament#5424] [Help] [answered]
Hello everyone,

Is it possible to have a Tabs on a ListResource page?

Let me explain: I have a Product table with the attributes `archived_at` and `deleted_at`, I would like to remove them from my main list while leaving the possibility for my user to consult the 'Archived' tab and the 'Deleted' tab.

I'm really having trouble finding my way through the documentation, and I feel like I'm completely missing out on some fabulous things you've developed.

Today I managed to meet my need by duplicating my Resource in 3 copies and playing with `getTableQuery()` by passing my query conditions there and adding at the top next to my 'Create' button other buttons that redirect to these pages. Yet they are the same views and the same data.

Thank you,

**Comment by danharrin** [comment]
We don't have an option for tabs at the moment, but it would be a cool addition. You can achieve something very similar with a table filter though

**Comment by ManojKiranA** [comment, marked as answer]
You don't need to create a different class for each tabs. I will explain the logic step by step

1. Declare a public variable called `$status` in the `ListRecords` page
2. Add the url of the list records actions along with status
3. When user clicks the button you must redirect to the list page along with status something like `domain.com/users?status=archived`
4. Change the query logic based on the `prop` in class by overriding `getTableQuery`

Something like this

 ```
 public $status;

  public function mount(): void
  {
      parent::mount();
      $this->status = request()->get('status');
  }

 public function getTableQuery(): Builder
  {
     return parent::getTableQuery()
          ->when($this->status, function ($query,$status) {
              return match($status){
                  'archived' => $query->whereNotNull('archived_at'),
                  default => $query,
              };
          });
  }
```

This above solution is just a workaround

and @danharrin It would be a nice addition to core itself becuase my workaround doesn't looks good

**Comment by ManojKiranA** [comment]
Hey @antoinemy its added to [roadmap](https://github.com/orgs/filamentphp/projects/2?pane=issue&itemId=17618975) 
Thanks for adding it @danharrin
